src/dialect/mod.rs

```diff
@@ -299,24 +330,33 @@ pub trait Dialect: Debug + Any {
         // return None to fall back to the default behavior
         None
     }
+
     /// Dialect-specific precedence override
+    ///
+    /// This method is called to get the precedence of the next token.
+    ///
+    /// If `None` is returned, falls back to the default behavior.
     fn get_next_precedence(&self, _parser: &Parser) -> Option<Result<u8, ParserError>> {
         // return None to fall back to the default behavior
         None
     }
 
-    /// Get the precedence of the next token. This "full" method means all precedence logic and remain
-    /// in the dialect. while still allowing overriding the `get_next_precedence` method with the option to
-    /// fallback to the default behavior.
+    /// Get the precedence of the next token, looking at the full token stream.
```

Comment:
Thanks for fixing this, sorry my comment was so broken.

I'm not sure "full token stream" is really the right distinction, my name was probably poor.

If we didn't might breaking the API, we could probably call the above `get_next_precedence_opt` - the only real difference is that it returns an `Option` so can fallback to the default behaviour.

Or we could call this method `get_next_precedence_default`.

Reply:
We haven't yet released this PR so we can change the name without any API breakages yet 

Any chance you can make a PR with a  proposed name change?

Reply:
@samuelcolvin  made a PR here: https://github.com/sqlparser-rs/sqlparser-rs/pull/1378